gax-httpjson/src/test/java/com/google/api/gax/httpjson/ProtoRestSerializerTest.java

```diff
@@ -0,0 +1,141 @@
+ * THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+package com.google.api.gax.httpjson;
+
+import com.google.common.truth.Truth;
+import com.google.protobuf.Field;
+import com.google.protobuf.Field.Cardinality;
+import com.google.protobuf.Option;
+import java.io.ByteArrayInputStream;
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import org.junit.Before;
+import org.junit.Test;
+
+public class ProtoRestSerializerTest {
+  private ProtoRestSerializer<Field> requestSerializer;
+  private Field field;
+  private String fieldJson;
+
+  @Before
+  public void setUp() {
+    requestSerializer = ProtoRestSerializer.create();
+    field =
+        Field.newBuilder()
+            .setNumber(2)
+            .setName("field_name1")
+            .addOptions(Option.newBuilder().setName("opt_name1").build())
+            .addOptions(Option.newBuilder().setName("opt_name2").build())
+            .setCardinality(Cardinality.CARDINALITY_OPTIONAL)
+            .build();
+
+    fieldJson =
+        "{\n"
+            + "  \"cardinality\": \"CARDINALITY_OPTIONAL\",\n"
+            + "  \"number\": 2,\n"
+            + "  \"name\": \"field_name1\",\n"
+            + "  \"options\": [{\n"
+            + "    \"name\": \"opt_name1\"\n"
+            + "  }, {\n"
+            + "    \"name\": \"opt_name2\"\n"
+            + "  }]\n"
+            + "}";
+  }
+
+  @Test
+  public void toJson() {
+    String fieldToJson = requestSerializer.toJson(field);
+    Truth.assertThat(fieldToJson).isEqualTo(fieldJson);
+  }
+
+  @Test
+  public void fromJson() {
+    Field fieldFromJson =
+        requestSerializer.fromJson(
+            new ByteArrayInputStream(fieldJson.getBytes(StandardCharsets.UTF_8)),
+            Field.newBuilder());
+
+    Truth.assertThat(fieldFromJson).isEqualTo(field);
+  }
+
+  @Test
+  public void fromJsonInvalidJson() {
+    try {
+      requestSerializer.fromJson(
+          new ByteArrayInputStream("heh".getBytes(StandardCharsets.UTF_8)), Field.newBuilder());
+      Truth.assertThat(true).isFalse();
+    } catch (RuntimeException e) {
+      Truth.assertThat(e.getCause()).isInstanceOf(IOException.class);
+    }
+  }
+
+  @Test
+  public void putPathParam() {
+    Map<String, String> fields = new HashMap<>();
+    requestSerializer.putPathParam(fields, "optName1", 1);
+    requestSerializer.putPathParam(fields, "optName2", 0);
+    requestSerializer.putPathParam(fields, "optName3", "three");
+    requestSerializer.putPathParam(fields, "optName4", "");
+
+    Map<String, String> expectedFields = new HashMap<>();
+    expectedFields.put("optName1", "1");
+    expectedFields.put("optName3", "three");
+
+    Truth.assertThat(fields).isEqualTo(expectedFields);
```

Comment:
What does it mean for `optName2` `optName4` to not be here? They're still settable by the app developer, correct?

Reply:
This tests for not putting default values. As we mentioned earlier, the whole default values thing will be revisited either way. This is what i needed to make MVP work.

Reply:
Ah, OK.